fulltext/factory.py

```diff
@@ -1,62 +1,135 @@
 """Application factory for fulltext service components."""
 
-import logging
-from flask import Flask
-from celery import Celery
-from fulltext import celeryconfig
-from fulltext.services import store, retrieve, fulltext, metrics
+import os
+import sys
+import logging as pylogging
+import re
+import time
 
-celery_app = Celery(__name__, results=celeryconfig.result_backend,
-                    broker=celeryconfig.broker_url)
-celery_app.config_from_object(celeryconfig)
-celery_app.autodiscover_tasks(['fulltext'], related_name='extract', force=True)
-celery_app.conf.task_default_queue = 'fulltext-worker'
+from typing import Any
+from typing_extensions import Protocol
+from werkzeug.exceptions import HTTPException, Forbidden, Unauthorized, \
+    BadRequest, MethodNotAllowed, InternalServerError, NotFound
+from werkzeug.routing import BaseConverter, ValidationError
+from flask import Flask, jsonify, Response
 
+from arxiv.base import Base, logging
+from arxiv.users.auth import Auth
+from arxiv.users import auth
+from arxiv.base.middleware import wrap, request_logs
 
-def create_web_app():
+from arxiv import vault
+
+from fulltext.services import store, pdf, compiler, extractor
+from fulltext.celery import celery_app
+from fulltext import routes
+from . import extract
+
+logger = logging.getLogger(__name__)
+
+
+class SubmissionSourceConverter(BaseConverter):
+    """Route converter for submission source identifiers."""
+
+    regex = r'[^/][0-9]+/[^/\?#]+'
+
+
+def create_web_app() -> Flask:
     """Initialize an instance of the web application."""
-    from fulltext.services import credentials
-    from fulltext import api
+    from . import celeryconfig
     app = Flask('fulltext')
     app.config.from_pyfile('config.py')
-    # logging.getLogger('boto').setLevel(logging.DEBUG)
-    # logging.getLogger('boto3').setLevel(logging.DEBUG)
-    # logging.getLogger('botocore').setLevel(logging.DEBUG)
-    credentials.init_app(app)
-    credentials.get_credentials()
-
-    app.register_blueprint(api.blueprint)
-    store.init_app(app)
-    retrieve.init_app(app)
-    fulltext.init_app(app)
-    metrics.init_app(app)
-
-    celery = Celery(app.name, results=celeryconfig.result_backend,
-                    broker=celeryconfig.broker_url)
-    celery.config_from_object(celeryconfig)
-    celery.autodiscover_tasks(['fulltext'], related_name='extract', force=True)
-    celery.conf.task_default_queue = 'fulltext-worker'
-    return app
+    celery_app.config_from_object(celeryconfig)
+    app.url_map.converters['source'] = SubmissionSourceConverter
+    pylogging.getLogger('boto').setLevel(pylogging.DEBUG)
```

Comment:
Are these log levels left over from debugging?

Reply:
oh yeah; should be more like this:

```python
    if app.config['LOGLEVEL'] < 40:
        # Make sure that boto doesn't spam the logs when we're in debug mode.
        pylogging.getLogger('boto').setLevel(pylogging.ERROR)
        pylogging.getLogger('boto3').setLevel(pylogging.ERROR)
        pylogging.getLogger('botocore').setLevel(pylogging.ERROR)
```

fixed in e080d53